test/validation.test.js

```diff
@@ -140,56 +224,96 @@ test(
 
 test(
   'Integer Test',
-  () =>
-    queryZcl
-      .selectAttributesByClusterCodeAndManufacturerCode(db, pkgId, 3, null)
-      .then((attribute) => {
-        attribute = attribute.filter((e) => {
-          return e.code === 0
-        })[0]
-
-        //Test Constraints
-        let minMax = validation.getBoundsInteger(attribute)
-        expect(minMax.min == 0).toBeTruthy()
-        expect(minMax.max === 0xffff).toBeTruthy()
-      }),
+  async () => {
+    let attribute =
+      await queryZcl.selectAttributesByClusterCodeAndManufacturerCode(
+        db,
+        pkgId,
+        3,
+        null
+      )
+    attribute = attribute.filter((e) => {
+      return e.code === 0
+    })[0]
+
+    const { size, isSigned } = await validation.getIntegerAttributeSize(
+      db,
+      sid,
+      3,
+      attribute.type
+    )
+    //Test Constraints
+    let minMax = await validation.getBoundsInteger(attribute, size, isSigned)
+    expect(minMax.min == 0).toBeTruthy()
+    expect(minMax.max === 0xffff).toBeTruthy()
+  },
   timeout.medium()
 )
 
 test(
   'validate Attribute Test',
-  () => {
+  async () => {
     let fakeEndpointAttribute = {
       defaultValue: '30',
     }
 
     let fakeAttribute = {
-      type: 'UINT16',
+      type: 'int16u',
```

Comment:
Why is this changed?

Reply:
while testing the queryZcl.selectNumberByNameAndClusterId() returned nothing with the upper case type name

my locally built zap tool also uses the lower case format whenever i edit an attribute, why was the other one used at the first place?

Reply:
How was this passing before?
The xml can vary how it represents unsigned 16 bit integers. I believe both representations exist based on matter/zigbee as of today.

Reply:
the validation did not check for specific integer types before my implementation, only decided if the input string is an integer and its signed value within bounds